polkadot/node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -1375,6 +1375,35 @@ async fn handle_network_msg<Context>(
 				unknown_heads: LruMap::new(ByLength::new(10)),
 			});
 
+			// Advertise collations for the current peer in case this is a reconnect.
+			//
+			// We might try to advertise collation T0 to the peer, then the peer disconnects
+			// before receiving the message. Later on, we generate a new collation T1
+			// and the peer reconnects. We need to make sure the peer gets T0 advertised as well.
+			//
+			// The `advertise_collation` ensures we are not readvertising the same collation
+			// multiple times.
+			if let Some(para_id) = state.collating_on {
```

Comment:
This should be handled by https://github.com/paritytech/polkadot-sdk/blob/c74ef8e8ef7ba2b03647c3b35727da8554a0bc49/polkadot/node/network/collator-protocol/src/collator_side/mod.rs#L1305?

Aka when the peer view is announced, it should be informed about the collations.